azure_scripts/lotka_volterra_francois.py

```diff
@@ -0,0 +1,251 @@
+  lotka_volterra_theta_bijector, get_samples_and_scores
+)
+from sbids.models import AffineSigmoidCoupling, ConditionalRealNVP
+
+os.makedirs("./outputs", exist_ok=True)
+
+# script arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("--batch_size", type=int, default=256)
+parser.add_argument("--n_simulations", type=int, default=500_000)
+parser.add_argument("--n_epochs", type=int, default=50)
+parser.add_argument("--dimension", type=int, default=4)
+parser.add_argument("--bijector_layers_size", type=int, default=64)
+parser.add_argument("--bijector_layers_shape", type=int, default=2)
+parser.add_argument("--nf_layers", type=int, default=4)
+parser.add_argument("--n_components", type=int, default=8)
+parser.add_argument("--score_weight", type=float, default=0.0)
+parser.add_argument("--model_seed", type=int, default=42)
+parser.add_argument("--initial_learning_rate", type=float, default=2e-3)
+parser.add_argument("--n_steps", type=int, default=20_000)
+args = parser.parse_args()
+
+if ON_AZURE:
+  run.log('batch_size', args.batch_size)
+  run.log('n_simulations', args.n_simulations)
+  run.log('n_epochs', args.n_epochs)
+  run.log('n_steps', args.n_steps)
+  run.log('dimension', args.dimension)
+  run.log('bijector_layers_size', args.bijector_layers_size)
+  run.log('bijector_layers_shape', args.bijector_layers_shape)
+  run.log('nf_layers', args.nf_layers)
+  run.log('n_components', args.n_components)
+  run.log('score_weight', args.score_weight)
+  run.log('model_seed', args.model_seed)
+  run.log('initial_learning_rate', args.initial_learning_rate)
+else:
+  print(args)
+
+rng_seq = hk.PRNGSequence(args.model_seed)
+
+# create truth
+task = sbibm.get_task("lotka_volterra")
+reference_samples = jnp.array(task.get_reference_posterior_samples(num_observation=1))
+observation = jnp.array(task.get_observation(num_observation=1).reshape([2, 10]).T) # Reordering for numpyro
+truth = jnp.array(task.get_true_parameters(num_observation=1).flatten())
+
+# create simulations
+@jax.jit
+def get_batch(key, batch_size=args.batch_size):
+    model = condition(lotka_volterra, {'z': jnp.array([30.,1.])})
+    (log_probs, samples), scores = get_samples_and_scores(model, key, batch_size=batch_size)
+    return samples['theta'], samples['y'].reshape([-1,20]), scores
+
+# create model
+bijector_layers = [args.bijector_layers_size] * args.bijector_layers_shape
+
+bijector = partial(
+  AffineSigmoidCoupling, 
+  layers=bijector_layers, 
+  n_components=args.n_components, 
+  activation=jnp.sin
+)
+
+NF = partial(
+  ConditionalRealNVP, 
+  n_layers=args.nf_layers, 
+  bijector_fn=bijector
+)
+
+class SmoothNPE(hk.Module):
+    def __call__(self, y):
+        net = lotka_volterra_y_bijector(y)
+        net = jnp.sin(hk.Linear(256, name='comp1')(net))
+        net = jnp.sin(hk.Linear(256, name='comp2')(net))
+        net = jnp.sin(hk.Linear(256, name='comp3')(net))
+        net = hk.Linear(args.n_components, name='comp4')(net)
```

Comment:
do we want to feed the conditional part of the NF with dimension equal to n_components ?

Reply:
That's a pure interpretation on my behalf, made when I saw on the notebook that the output layer of the MLP was the same size as the number of components. 
But that may be a completely wrong assertion, in which case we should use a different argument value for it.

Reply:
I don't know ^^ @EiffL ?

Reply:
No not n_conponents but typically the output of your compressor is of the same size (or same order of magnitude) as the number of dimensions of the distribution you are trying to estimate